image/psd.ksy

```diff
@@ -0,0 +1,574 @@
+meta:
+  id: psd
+  title: Photoshop PSD
+  file-extension: psd
+  license: CC0-1.0
+  ks-version: 0.9
+  endian: be
+doc-ref: https://www.adobe.com/devnet-apps/photoshop/fileformatashtml/
+seq:
+  - id: header
+    type: header
+  - id: color_mode_data
+    type: color_mode_data
+  - id: image_resources
+    type: image_resources_data
+  - id: layer_and_mask_information
+    type: layer_and_mask_information
+  - id: image_data
+    type: image_data
+types:
+  header:
+    seq:
+      - id: magic
+        contents: "8BPS"
+      - id: version
+        type: u2
+        valid: 1
+      - id: reserved
+        contents: [0x00, 0x00, 0x00, 0x00, 0x00, 0x00]
+      - id: num_channels
+        type: u2
+        valid:
+          min: 1
+          max: 56
+      - id: height
+        type: u4
+        valid:
+          min: 1
+          max: 30000
+      - id: width
+        type: u4
+        valid:
+          min: 1
+          max: 30000
+      - id: depth
+        type: u2
+        valid:
+          any-of: [1, 8, 16, 32]
+      - id: color_mode
+        type: u2
+        enum: color_mode
+        valid:
+          any-of:
+            - color_mode::bitmap
+            - color_mode::grayscale
+            - color_mode::indexed
+            - color_mode::rgb
+            - color_mode::cmyk
+            - color_mode::multichannel
+            - color_mode::duotone
+            - color_mode::lab
+  color_mode_data:
+    seq:
+      - id: len_color_mode
+        type: u4
+      - id: color_mode
+        size: len_color_mode
+  image_resources_data:
+    seq:
+      - id: len_resources
+        type: u4
+      - id: resources
+        size: len_resources
+        type: image_resources
+  image_resources:
+    seq:
+      - id: image_resources
+        type: image_resource
+        repeat: eos
+  image_resource:
+    -webide-representation: '{resource_id}'
+    seq:
+      - id: magic
+        contents: "8BIM"
+      - id: resource_id
+        type: u2
+        enum: resource_ids
+      - id: name
+        type: pascal_string
+      - id: padding1
+        size: 1
+        if: name.len_string % 2 == 0
```

Comment:
Is padding should be a function of `len_string`, not `_io.pos`?

Reply:
No, not for `_io.pos`. The specification says: "Name: Pascal string, padded to make the size even (a null name consists of two bytes of 0)".